seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/client/EsRestClient.java

```diff
@@ -599,6 +618,14 @@ private static Map<String, BasicTypeDefine<EsType>> getFieldTypeMappingFromPrope
                                 typeDefine.nativeType(
                                         new EsType(OBJECT, (Map) subFieldTypeInfoMap));
                                 allElasticSearchFieldTypeInfoMap.put(fieldName, typeDefine.build());
+                            } else {
+                                // For map types
+                                BasicTypeDefine.BasicTypeDefineBuilder<EsType> typeDefine =
+                                        BasicTypeDefine.<EsType>builder()
+                                                .name(fieldName)
+                                                .columnType("text")
+                                                .dataType("text");
+                                allElasticSearchFieldTypeInfoMap.put(fieldName, typeDefine.build());
```

Comment:
Quick quesiton: how to create a map type in elasticsearch?

Reply:
The previous object type conversion exception added this judgment, which has now been removed